include/zephyr/dt-bindings/pwm/stm32_pwm.h

```diff
@@ -16,10 +16,12 @@
  * @{
  */
 /** PWM complementary output pin is enabled */
-#define PWM_STM32_COMPLEMENTARY	(1U << 8)
+#define PWM_STM32_COMPLEMENTARY	(1U << 8) __DEPRECATED_MACRO
```

Comment:
`__DEPRECATED_MACRO` will not work for definitions used in devicetree.

Reply:
Arf, so what are the guidelines to deprecate dts bindings ?